src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -2,15 +2,41 @@
 using System.Buffers;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using System.Threading;
+using Sparrow.Collections;
 using Sparrow.LowMemory;
 using Sparrow.Threading;
 using Sparrow.Utils;
 
 namespace Sparrow.Json
 {
+    public class ThreadIdHolder
+    {
+        public int ThreadId;
+        public override string ToString()
+        {
+            return ThreadId.ToString();
+        }
+    }
+
+#if DEBUG
+    public interface ThreadIDsHolderInterface
```

Comment:
I don't like all the code associated with it.
I don't think we need to carry it forward

Reply:
I would actually feel more comfortable if we'll be running with this "debug" code for a while, at least until being a bit closer to 4.2 release

Reply:
@ayende ?

Reply:
No, that is really heavy weight and it is something that we know will cause us issues.
We can always bring back this code

Reply:
ok, created an issue for that, we'll let a few more nightlies to run with that and we'll get it removed https://issues.hibernatingrhinos.com/issue/RavenDB-12699